internal/weapons/sword/travelershandysword/travelershandysword.go

```diff
@@ -0,0 +1,32 @@
+package travelershandysword
+
+import (
+	"fmt"
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+func init() {
+	core.RegisterWeaponFunc("travelershandysword", weapon)
+}
+
+// Each Elemental Orb or Particle collected restores 1/1.25/1.5/1.75/2% HP.
+func weapon(char core.Character, c *core.Core, r int, param map[string]int) string {
+	c.Events.Subscribe(core.OnParticleReceived, func(args ...interface{}) bool {
+		// ignore if character not on field
+		if c.ActiveChar != char.CharIndex() {
+			return false
+		}
+
+		c.Health.Heal(core.HealInfo{
+			Caller:  char.CharIndex(),
+			Target:  c.ActiveChar,
+			Message: "Traveler's Handy Sword (Proc)",
+			Src:     char.MaxHP() * (0.0075 + float64(r)*0.0025),
```

Comment:
there's a type `HealTypePercent` that heals the char from his max hp
```go
Type:    core.HealTypePercent,
Message: "Traveler's Handy Sword (Proc)",
Src:     0.0075 + float64(r)*0.0025,
```

Reply:
@imring TYVM, that's so much easier!